### PR TITLE
also show message for empty display name if change of it isn't allowed

### PR DESCRIPTION
**Pull request by MorrisJobke**
cc @DeepDiver1975 @LukasReschke 

fixes https://github.com/owncloud/core/pull/13395#discussion_r23149066

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
:+1:

File: settings/templates/personal.php [hunk 1]
```diff
@@ -130,7 +130,7 @@
 ?>
 <div class="section">
 	<h2><?php echo $l->t('Email'); ?></h2>
-	<span><?php if($_['email']) { p($_['email']); } else { p($l->t('No email address set')); }?></span>
+	<span><?php if(empty($_['email'])) { p($_['email']); } else { p($l->t('No email address set')); }?></span>
```
Comment (nickvergessen) [inline settings/templates/personal.php]: `!empty()`

File: settings/templates/personal.php [hunk 2]
```diff
@@ -106,7 +106,7 @@
 ?>
 <div class="section">
 	<h2><?php echo $l->t('Full Name');?></h2>
-	<span><?php p($_['displayName'])?></span>
+	<span><?php if(empty($_['displayName'])) { p($_['displayName']); } else { p($l->t('No display name set')); } ?></span>
```
Comment (nickvergessen) [inline settings/templates/personal.php]: `!empty()`

**Comment by nickvergessen** [comment]
:-1:

**Comment by MorrisJobke** [comment]
:( Sorry

**Comment by MorrisJobke** [comment]
Fixed :)

**Comment by MorrisJobke** [comment]
And thanks for spotting

**Comment by nickvergessen** [comment]
Display name `0` is not considered valid I guess?

:+1: if it is not valid

**Comment by MorrisJobke** [comment]
> Display name 0 is not considered valid I guess?

Should I add an exception for that one?

**Comment by nickvergessen** [comment]
Since this is only used when the user can not change the display name, I'd say it's fine as is.
No sane admin will use `0` for a display name of a user

**Comment by LukasReschke** [comment]
We could otherwise also use `isset($foo[0])` here which will also check for a not empty string

**Comment by nickvergessen** [comment]
I guess that works aswell @MorrisJobke ?

**Comment by MorrisJobke** [comment]
Nope. Because it is set with an empty string. I will just check for `''`.

**Comment by nickvergessen** [comment]
isset() on `[0]` of an empty string is false ?!

**Comment by MorrisJobke** [comment]
ah ...now I got it :D

**Comment by MorrisJobke** [comment]
Updated

**Comment by nickvergessen** [comment]
:+1: again

**Pull request merged**